nupic/research/frameworks/pytorch/test_utils/fake_data.py

```diff
@@ -18,15 +18,50 @@
 # http://numenta.org/licenses/
 # ----------------------------------------------------------------------
 
+import torch
 from torch.utils.data import DataLoader
+from torchvision import transforms
 from torchvision.datasets import FakeData
-from torchvision.transforms import ToTensor
 
 __all__ = [
+    "FakeDataPredefinedTargets",
     "FakeDataLoader",
 ]
 
 
+class FakeDataPredefinedTargets(FakeData):
```

Comment:
Why is this necessary? Why do you need predefined `targets` for fake data?

Reply:
because the `split_dataset` function assumes the dataset class has a `targets attribute` (which MNIST does), but the `FakeData` class does not.

Reply:
This is a new code to `split_dataset` regarding target_transform. If you apply target_transform before splitting and restore `split_dataset` to its original version, you won't need to access `targets` from ``split_dataset` and will be able to use `FakeData`